emission/tests/modellingTests/TestSimilarityMetric.py

```diff
@@ -1,31 +1,59 @@
 import unittest
-import emission.tests.modellingTests.modellingTestAssets as etmm
 import emission.analysis.modelling.similarity.od_similarity as eamso
-
+import emission.tests.modellingTests.modellingTestAssets as etmm
 class TestSimilarityMetric(unittest.TestCase):
 
     def testODsAreSimilar(self):
-        generate_points_thresh = 0.001  # approx. 111 meters
-        similarity_threshold = 500  # 
-        # random, but, points are sampled within a circle and should always be < sim threshold
-        trips = etmm.generate_mock_trips('bob', 2, [0, 0], [1, 1], threshold=generate_points_thresh)
+        similarity_threshold = 500  # in meters
         metric = eamso.OriginDestinationSimilarity()
-        coords0 = metric.extract_features(trips[0])
-        coords1 = metric.extract_features(trips[1])
-        similar = metric.similar(coords0, coords1, similarity_threshold)
-        self.assertTrue(similar)
+
+        # parameters passed for testing is set here. A list, where each element of this list takes the form 
+        # [trip part to be sampled within mentioned threshold, (start_coord,end_coord)]
+        # Since the extracted_features function returns in the form [origin_lat,origin_long,destination_lat,destination_long],
+        # if clustering is to be done by :
+        #   a.origin, we pass first two values of this list,i.e. from 0 till before 2 index
+        #   b.destination, we pas last two values of this list,i.e. from 2 till before 4 index
+        #   c.origin-destination, we pass the entire list , i.e. from 0 till before 4 index
+        parameters= [["o_",'origin'],["_d",'destination'],["od",'origin-destination']]
+
+        for tp,cw in parameters:
+            with self.subTest(trip_part=tp):
+                #generate 2 trips with parameter values
+                trips = etmm.generate_mock_trips('joe',2, threshold=similarity_threshold,origin=[0, 0], destination=[1, 1], within_threshold=2,trip_part=tp) 
+                # depending on the parametrs, extract the relevant coordinates
+                trip0_coords = metric.extract_features(trips[0])
+                trip1_coords = metric.extract_features(trips[1])
+                #check for similarity using relevant coordinates
+                similarOD = metric.similar(trip0_coords,trip1_coords, similarity_threshold,cw)
+                # Since both origin and destination poitns lie within threshold limits,they should be similar
+                # when we check by just origin or just destination or both origin-and-destination
+                self.assertTrue(similarOD)
```

Comment:
nit: **(future fix)** if this fails, there is no message which indicates why. and given that it is an `assertTrue`, you won't even get the autogenerated message.